contracts/token/erc1155/abstract/ERC1155Permit.Sol

```diff
@@ -0,0 +1,159 @@
+            recoveredSigner = ECDSA.recover(
+                digest,
+                bytes32(BytesLib.slice(sig, 0, 32)),
+                bytes32(BytesLib.slice(sig, 32, 64))
+            );
+        } else if (sig.length == 65) {
+            // typical EDCSA Sig
+            recoveredSigner = ECDSA.recover(digest, sig);
+        } else {
+            revert InvalidSignature();
+        }
+
+        if (_isValidEOASignature(recoveredSigner, owner)) {
+            _setApprovalForAll(owner, spender, approved);
+        } else {
+            revert InvalidSignature();
+        }
+    }
+
+    /**
+     * @notice Builds the EIP-712 compliant digest for the permit.
+     * @param spender The address which is approved to spend the token.
+     * @param owner The address that holds the tokens.
+     * @param deadline The deadline until which the permit is valid.
+     * @return A bytes32 digest, EIP-712 compliant, that serves as a unique identifier for the permit.
+     */
+    function _buildPermitDigest(
+        address spender,
+        address owner,
+        bool approved,
+        uint256 deadline
+    ) internal returns (bytes32) {
+        return _hashTypedDataV4(
+            keccak256(
+                abi.encode(
+                    _PERMIT_TYPEHASH,
+                    owner,
+                    spender,
+                    approved,
+                    _nonces[owner]++,
+                    deadline
+                )
+            )
+        );
+    }
+
+    /**
+     * @notice Checks if a given signature is valid according to EIP-1271.
+     * @param spender The address which purports to have signed the message.
+     * @param digest The EIP-712 compliant digest that was signed.
+     * @param sig The actual signature bytes.
+     * @return True if the signature is valid according to EIP-1271, otherwise false.
+     */
+    function _isValidERC1271Signature(address spender, bytes32 digest, bytes memory sig) private view returns(bool) {
+        (bool success, bytes memory res) = spender.staticcall(
+            abi.encodeWithSelector(
+                IERC1271.isValidSignature.selector, 
+                digest, 
+                sig
+            )
+        );
+
+        if (success && res.length == 32) {
+            bytes4 decodedRes = abi.decode(res, (bytes4));
+            if (decodedRes == IERC1271.isValidSignature.selector) {
+                return true;
+            }
+        }
+        
+        return false;
+    }
+
+    /**
+     * @notice Checks if a given signature is valid according to EIP-1271.
+     * @param recoveredSigner The address which purports to have signed the message.
+     * @param owner The owner of the tokens.
+     * @return True if the signature is from an approved operator or owner, otherwise false.
+     */
+    function _isValidEOASignature(address recoveredSigner, address owner) private pure returns(bool) {
+        return recoveredSigner != address(0) && recoveredSigner == owner;
```

Comment:
Should this check if `recoveredSigner` has approval?

Reply:
we could use `isApprovedforAll` but i think thats a bit to dangerous to have someone with perm for all of your tokens to then give someone else perms to all of your tokens.